opmon/cli.py

```diff
@@ -377,18 +401,121 @@ def backfill(
             )
             monitoring.run(date)
         except Exception as e:
             print(f"Error backfilling {config[0]}: {e}")
             success = False
 
-    Metadata(project_id, dataset_id, [config]).write()
+    Metadata(project_id, dataset_id, derived_dataset_id, [config]).write()
 
     sys.exit(0 if success else 1)
 
 
+@cli.command()
+@click.pass_context
+@click.option(
+    "--project_id",
+    "--project-id",
+    default="mozdata",
+    help="Project to write to",
+)
+@click.option(
+    "--dataset_id",
+    "--dataset-id",
+    default="tmp",
+    help="Temporary dataset to write to",
+)
+@click.option(
+    "--derived_dataset_id",
+    "--derived-dataset-id",
+    default="tmp",
+    help="Temporary derived dataset to write to",
+)
+@click.option(
+    "--start_date",
+    "--start-date",
+    type=ClickDate(),
+    help="Date for which project should be started to get analyzed. Default: current date - 3 days",
+    metavar="YYYY-MM-DD",
+    required=False,
+)
+@click.option(
+    "--end_date",
+    "--end-date",
+    type=ClickDate(),
+    help="Date for which project should be stop to get analyzed. Default: current date",
+    metavar="YYYY-MM-DD",
+    required=False,
+)
+@click.option(
+    "--slug",
+    help="Experimenter or Normandy slug associated with the project to create a preview for",
+    required=True,
+)
+@click.option(
+    "--config_file",
+    "--config-file",
+    help="Custom local config file",
+    required=False,
+    type=click.Path(exists=True),
+)
+@config_repos_option
+@private_config_repos_option
+def preview(
+    ctx,
+    project_id,
+    dataset_id,
+    derived_dataset_id,
+    start_date,
+    end_date,
+    slug,
+    config_file,
+    config_repos,
+    private_config_repos,
+):
+    """Create a preview for a specific project based on a subset of data."""
+    if start_date is None and end_date is None:
+        start_date = datetime.today() - timedelta(days=3)
+        end_date = datetime.today()
+    elif start_date is None:
+        start_date = end_date - timedelta(days=3)
+    else:
+        end_date = start_date + timedelta(days=3)
+
+    start_date = pytz.utc.localize(start_date)
+    end_date = pytz.utc.localize(end_date)
+
+    start_date_str = start_date.strftime("%Y-%m-%d")
+    end_date_str = end_date.strftime("%Y-%m-%d")
+    table = bq_normalize_name(slug)
+
+    click.echo(
+        "A preview is available at: "
+        + "https://mozilla.cloud.looker.com/dashboards/975?"
+        + f'Table="{project_id}.{dataset_id}.{table}"'
+        + f"&Submission+Date={start_date_str}+to+{end_date_str}"
+    )
+
+    ctx.invoke(
+        backfill,
+        project_id=project_id,
+        dataset_id=dataset_id,
+        derived_dataset_id=derived_dataset_id,
+        start_date=start_date,
+        end_date=end_date,
+        slug=slug,
+        config_file=config_file,
+        config_repos=config_repos,
+        private_config_repos=private_config_repos,
+    )
```

Comment:
We could be more clever about only running this ETL on a sample of data, but this might require more time in terms of refactoring the code here than the money we would actually save. For now, it's probably enough to only run this on 3 days by default